src/main/java/com/bgsoftware/superiorskyblock/core/io/MenuParserImpl.java

```diff
@@ -399,10 +401,65 @@ private static TemplateItem getItemStackUnsafe(String fileName, ConfigurationSec
             }
         }
 
+        if (section.contains("bannerMeta")) {
+            for (String _dyeColor : section.getConfigurationSection("bannerMeta").getKeys(false)) {
+                DyeColor dyeColor;
+                PatternType patternType;
+
+                try {
+                    dyeColor = DyeColor.valueOf(_dyeColor.toUpperCase(Locale.ENGLISH));
+                } catch (IllegalArgumentException error) {
+                    Log.warnFromFile(fileName, "Couldn't convert ", section.getCurrentPath(),
+                            ".bannerMeta.", _dyeColor.toUpperCase(Locale.ENGLISH), " into an dye color, skipping...");
+                    continue;
+                }
+
+                try {
+                    patternType = PatternType.valueOf(section.getString("bannerMeta." + _dyeColor));
+                } catch (IllegalArgumentException error) {
+                    Log.warnFromFile(fileName, "Couldn't convert ", section.getCurrentPath(),
+                            ".bannerMeta.", _dyeColor.toUpperCase(Locale.ENGLISH), ".",
+                            section.getString("bannerMeta." + _dyeColor), " into an pattern type, skipping...");
+                    continue;
+                }
+
+                itemBuilder.withBannerMeta(dyeColor, patternType);
+            }
+        }
+
         if (section.contains("customModel")) {
             itemBuilder.withCustomModel(section.getInt("customModel"));
         }
 
+        if (section.contains("itemModel")) {
+            itemBuilder.withItemModel(section.getString("itemModel"));
+        }
+
+        if (section.contains("rarity")) {
+            String rarity = section.getString("rarity");
+
+            try {
+                itemBuilder.withRarity(rarity);
+            } catch (IllegalArgumentException error) {
+                Log.warnFromFile(fileName, "Couldn't convert ", rarity, " into a rarity, skipping...");
+            }
+        }
+
+        if (section.contains("trimMaterial") || section.contains("trimPattern")) {
+            String trimMaterial = section.getString("trimMaterial");
+            String trimPattern = section.getString("trimPattern");
+
+            if (trimMaterial != null && trimPattern != null) {
+                itemBuilder.withTrim(trimMaterial, trimPattern);
```

Comment:
There's no warning if trimMaterial or trimPattern are invalid, as the setTrim function does not throw any error in such case.